src/transpiling/mod.rs

```diff
@@ -146,41 +163,115 @@ impl TranspileOptions {
 
 impl ParsedSource {
   /// Transform a TypeScript file into a JavaScript file.
+  ///
+  /// Note: This will clone the program if it's shared, which
+  /// might be expensive.
   pub fn transpile(
```

Comment:
We should probably get rid of this in the future and make it internal. I just wanted to do this in a backwards compatible way for the time being.